app/components/channel_drawer/channels_list/channels_list.js

```diff
@@ -0,0 +1,406 @@
+// Copyright (c) 2016-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import React, {Component} from 'react';
+import PropTypes from 'prop-types';
+import {
+    Platform,
+    Text,
+    TouchableHighlight,
+    View
+} from 'react-native';
+import {injectIntl, intlShape} from 'react-intl';
+import AwesomeIcon from 'react-native-vector-icons/FontAwesome';
+import MaterialIcon from 'react-native-vector-icons/MaterialIcons';
+
+import Badge from 'app/components/badge';
+import SearchBar from 'app/components/search_bar';
+import {preventDoubleTap} from 'app/utils/tap';
+import {changeOpacity, makeStyleSheetFromTheme} from 'app/utils/theme';
+
+import FilteredList from './filtered_list';
+import List from './list';
+
+class ChannelsList extends Component {
```

Comment:
Shouldn't this be a PureComponent?

Reply:
It was just a Component before, but I can't see any reason it isn't pure. I'll change it